inc/class-provider.php

```diff
@@ -44,40 +46,60 @@ public function __construct( Factory $factory ) {
 	 *
 	 * @throws Exception If the REST API response could not be decoded.
 	 */
-	protected function request( array $args ): MediaList {
-
-		$args = $this->parse_args( $args );
-
-		$url = get_endpoint();
-		$url = add_query_arg( $args, $url );
-
-		$response = $this->remote_request( $url, [
-			'headers' => [
-				'Accept-Encoding' => 'gzip, deflate',
-				'Connection'      => 'Keep-Alive',
-				'Content-Type'    => 'application/json',
-				'Keep-Alive'      => 30,
-			],
-			'timeout' => 30,
-		] );
-		$response = json_decode( $response );
-
-		if ( json_last_error() ) {
-			throw new Exception( sprintf(
-				/* translators: %s: Error message */
-				__( 'Media error: %s', 'amf-wordpress' ),
-				json_last_error_msg()
-			) );
-		}
-
-		if ( ! is_array( $response ) || ! $response ) {
-			return new MediaList();
-		}
-
-		$items = array_map( [ $this->factory, 'create' ], $response );
-
-		return new MediaList( ...$items );
-	}
+    protected function request( array $args ): MediaList
+    {
+        $args = $this->parse_args( $args );
+
+        // if this is a local multisite, run a query instead of hitting the external API.
+        if ( defined( 'AMF_LOCAL_BLOG_ID' ) {
+			  $local_blog_id = AMF_LOCAL_BLOG_ID;
+            $current_blog = get_current_blog_id();
+
+            switch_to_blog( $local_blog_id );
+
+            $controller = new WP_REST_Attachments_Controller( 'attachment' );
+            $request = new WP_REST_Request( 'GET', '/wp/v2/media', $controller->get_collection_params() );
+            $request->set_query_params( $args );
+            $response = $controller->get_items( $request );
+
+            if ( !empty( $response ) && isset( $response->data ) ) {
+                $response = array_map(function ( $item ) {
+					return json_decode( json_encode( $item ), false );
+                }, $response->data);
+            }
+
+            switch_to_blog( $current_blog );
```

Comment:
You can use `restore_current_blog()` instead, then you don't need the `$current_blog` variable.